contracts/token/SRC20.sol

```diff
@@ -108,21 +135,25 @@ contract SRC20 is ISRC20, ISRC20Owned, ISRC20Managed, SRC20Detailed, Featured,
     function _updateKYA(bytes32 kyaHash, string memory kyaUrl, address restrictions) internal returns (bool) {
         _kya.kyaHash = kyaHash;
         _kya.kyaUrl = kyaUrl;
-        _restrictions = ITransferRestriction(restrictions);
 
-        emit KYAUpdated(kyaHash, kyaUrl, restrictions);
+        _restrictions = ITransferRules(restrictions);
+
+        if (restrictions != address(0)) {
+            _restrictions.setSRC(address(this));
```

Comment:
Again, if only SRC20 will call restriction contract, this is not needed. If it stays, it should be under `require` as it returns bool.